packages/text-annotator/src/SelectionHandler.ts

```diff
@@ -147,13 +171,14 @@ export const SelectionHandler = (
       }
     }
 
+    const sel = document.getSelection();
     const timeDifference = evt.timeStamp - lastPointerDown.timeStamp;
 
     // Just a click, not a selection
-    if (document.getSelection().isCollapsed && timeDifference < 300) {
+    if (sel?.isCollapsed && timeDifference < CLICK_TIMEOUT) {
       currentTarget = undefined;
       clickSelect();
-    } else if (currentTarget) {
+    } else if (currentTarget && store.getAnnotation(currentTarget.annotation)) {
       selection.userSelect(currentTarget.annotation, evt);
     }
```

Comment:
Prevents selection of the `currentTarget` that could have already been deleted in the `selectionchange` event handler.